[Fabric]Build fail on react-native@0.69.12

### Description

`lottie-react-native@6.0.0-rc.7` build fail on `react-native@0.69.12` with `newArchEnabled=true`

### Steps to Reproduce

```sh
$ npx react-native@0.69.12 init MyReactNative --version 0.69.12
$ cd MyReactNative && yarn add lottie-react-native@6.0.0-rc.7
$ yarn start
$ ORG_GRADLE_PROJECT_newArchEnabled=true yarn android
```

**Expected behavior:** Build success

**Actual behavior:** Build fail

```
ERROR:/Users/sunbreak/w/Sunbreak/reactnative-project.trial/MyReactNative@0.69.12/node_modules/lottie-react-native/android/build/.transforms/9cb7e324d36f41a97d51360bd31ff4ea/transformed/classes/classes.dex: D8: Type com.facebook.react.viewmanagers.ActivityIndicatorViewManagerInterface is defined multiple times: /Users/sunbreak/w/Sunbreak/reactnative-project.trial/MyReactNative@0.69.12/node_modules/lottie-react-native/android/build/.transforms/9cb7e324d36f41a97d51360bd31ff4ea/transformed/classes/classes.dex, /Users/sunbreak/w/Sunbreak/reactnative-project.trial/MyReactNative@0.69.12/node_modules/react-native/ReactAndroid/build/.transforms/786208841fda84dcdbade4bdeabb5844/transformed/classes/classes.dex
com.android.builder.dexing.DexArchiveMergerException: Error while merging dex archives: 
Type com.facebook.react.viewmanagers.ActivityIndicatorViewManagerInterface is defined multiple times: /Users/sunbreak/w/Sunbreak/reactnative-project.trial/MyReactNative@0.69.12/node_modules/lottie-react-native/android/build/.transforms/9cb7e324d36f41a97d51360bd31ff4ea/transformed/classes/classes.dex, /Users/sunbreak/w/Sunbreak/reactnative-project.trial/MyReactNative@0.69.12/node_modules/react-native/ReactAndroid/build/.transforms/786208841fda84dcdbade4bdeabb5844/transformed/classes/classes.dex
Learn how to resolve the issue at https://developer.android.com/studio/build/dependencies#duplicate_classes.
        at com.android.builder.dexing.D8DexArchiveMerger.getExceptionToRethrow(D8DexArchiveMerger.java:151)
        at com.android.builder.dexing.D8DexArchiveMerger.mergeDexArchives(D8DexArchiveMerger.java:138)
        at com.android.build.gradle.internal.tasks.DexMergingWorkAction.merge(DexMergingTask.kt:859)
        at com.android.build.gradle.internal.tasks.DexMergingWorkAction.run(DexMergingTask.kt:805)
        at com.android.build.gradle.internal.profile.ProfileAwareWorkAction.execute(ProfileAwareWorkAction.kt:74)
        at org.gradle.workers.internal.DefaultWorkerServer.execute(DefaultWorkerServer.java:63)
        at org.gradle.workers.internal.NoIsolationWorkerFactory$1$1.create(NoIsolationWorkerFactory.java:66)
        at org.gradle.workers.internal.NoIsolationWorkerFactory$1$1.create(NoIsolationWorkerFactory.java:62)
        at org.gradle.internal.classloader.ClassLoaderUtils.executeInClassloader(ClassLoaderUtils.java:97)
        at org.gradle.workers.internal.NoIsolationWorkerFactory$1.lambda$execute$0(NoIsolationWorkerFactory.java:62)
        at org.gradle.workers.internal.AbstractWorker$1.call(AbstractWorker.java:44)
        at org.gradle.workers.internal.AbstractWorker$1.call(AbstractWorker.java:41)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:204)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:199)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:66)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:59)
        at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:157)
        at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:59)
        at org.gradle.internal.operations.DefaultBuildOperationRunner.call(DefaultBuildOperationRunner.java:53)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.call(DefaultBuildOperationExecutor.java:73)
        at org.gradle.workers.internal.AbstractWorker.executeWrappedInBuildOperation(AbstractWorker.java:41)
        at org.gradle.workers.internal.NoIsolationWorkerFactory$1.execute(NoIsolationWorkerFactory.java:59)
        at org.gradle.workers.internal.DefaultWorkerExecutor.lambda$submitWork$2(DefaultWorkerExecutor.java:206)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at org.gradle.internal.work.DefaultConditionalExecutionQueue$ExecutionRunner.runExecution(DefaultConditionalExecutionQueue.java:214)
        at org.gradle.internal.work.DefaultConditionalExecutionQueue$ExecutionRunner.runBatch(DefaultConditionalExecutionQueue.java:164)
        at org.gradle.internal.work.DefaultConditionalExecutionQueue$ExecutionRunner.run(DefaultConditionalExecutionQueue.java:131)
        at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
        at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:48)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:61)
        at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: com.android.tools.r8.CompilationFailedException: Compilation failed to complete, origin: /Users/sunbreak/w/Sunbreak/reactnative-project.trial/MyReactNative@0.69.12/node_modules/lottie-react-native/android/build/.transforms/9cb7e324d36f41a97d51360bd31ff4ea/transformed/classes/classes.dex
        at Version.fakeStackEntry(Version_3.1.51.java:0)
        at com.android.tools.r8.internal.oO0.a(SourceFile:68)
        at com.android.tools.r8.internal.oO0.a(SourceFile:28)
        at com.android.tools.r8.internal.oO0.a(SourceFile:27)
        at com.android.tools.r8.internal.oO0.b(SourceFile:3)
        at com.android.tools.r8.D8.run(D8.java:11)
        at com.android.builder.dexing.D8DexArchiveMerger.mergeDexArchives(D8DexArchiveMerger.java:136)
        ... 33 more
Caused by: com.android.tools.r8.internal.b: Type com.facebook.react.viewmanagers.ActivityIndicatorViewManagerInterface is defined multiple times: /Users/sunbreak/w/Sunbreak/reactnative-project.trial/MyReactNative@0.69.12/node_modules/lottie-react-native/android/build/.transforms/9cb7e324d36f41a97d51360bd31ff4ea/transformed/classes/classes.dex, /Users/sunbreak/w/Sunbreak/reactnative-project.trial/MyReactNative@0.69.12/node_modules/react-native/ReactAndroid/build/.transforms/786208841fda84dcdbade4bdeabb5844/transformed/classes/classes.dex
        at com.android.tools.r8.internal.il1.a(SourceFile:14)
        at com.android.tools.r8.internal.il1.a(SourceFile:22)
        at com.android.tools.r8.internal.lf1.b(SourceFile:6)
        at com.android.tools.r8.internal.lf1.a(SourceFile:23)
        at com.android.tools.r8.internal.lf1.a(SourceFile:10)
        at java.base/java.util.concurrent.ConcurrentHashMap.merge(ConcurrentHashMap.java:2048)
        at com.android.tools.r8.internal.lf1.a(SourceFile:6)
        at com.android.tools.r8.graph.r2$a.f(SourceFile:4)
        at com.android.tools.r8.dex.b.a(SourceFile:96)
        at com.android.tools.r8.dex.b.a(SourceFile:21)
        at com.android.tools.r8.D8.a(D8.java:21)
        at com.android.tools.r8.D8.d(D8.java:11)
        at com.android.tools.r8.D8.b(D8.java:1)
        at com.android.tools.r8.internal.oO0.a(SourceFile:24)
        ... 36 more


FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:mergeLibDexDebug'.
> A failure occurred while executing com.android.build.gradle.internal.tasks.DexMergingTaskDelegate
   > There was a failure while executing work items
      > A failure occurred while executing com.android.build.gradle.internal.tasks.DexMergingWorkAction
         > com.android.builder.dexing.DexArchiveMergerException: Error while merging dex archives: 
           Type com.facebook.react.viewmanagers.ActivityIndicatorViewManagerInterface is defined multiple times: /Users/sunbreak/w/Sunbreak/reactnative-project.trial/MyReactNative@0.69.12/node_modules/lottie-react-native/android/build/.transforms/9cb7e324d36f41a97d51360bd31ff4ea/transformed/classes/classes.dex, /Users/sunbreak/w/Sunbreak/reactnative-project.trial/MyReactNative@0.69.12/node_modules/react-native/ReactAndroid/build/.transforms/786208841fda84dcdbade4bdeabb5844/transformed/classes/classes.dex
           Learn how to resolve the issue at https://developer.android.com/studio/build/dependencies#duplicate_classes.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 11m 38s

    at makeError (/Users/sunbreak/w/Sunbreak/reactnative-project.trial/MyReactNative@0.69.12/node_modules/execa/index.js:174:9)
    at /Users/sunbreak/w/Sunbreak/reactnative-project.trial/MyReactNative@0.69.12/node_modules/execa/index.js:278:16
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async runOnAllDevices (/Users/sunbreak/w/Sunbreak/reactnative-project.trial/MyReactNative@0.69.12/node_modules/@react-native-community/cli-platform-android/build/commands/runAndroid/runOnAllDevices.js:109:5)
    at async Command.handleAction (/Users/sunbreak/w/Sunbreak/reactnative-project.trial/MyReactNative@0.69.12/node_modules/@react-native-community/cli/build/index.js:192:9)
info Run CLI with --verbose flag for more details.
error Command failed with exit code 1.
```

### Versions

- npx react-native@0.69.12 info

```
info Fetching system and libraries information...
System:
    OS: macOS 12.6.7
    CPU: (8) arm64 Apple M1 Pro
    Memory: 468.52 MB / 32.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 18.16.1 - ~/.volta/tools/image/node/18.16.1/bin/node
    Yarn: 1.22.19 - ~/.volta/tools/image/yarn/1.22.19/bin/yarn
    npm: 9.5.1 - ~/.volta/tools/image/node/18.16.1/bin/npm
    Watchman: Not Found
  Managers:
    CocoaPods: 1.12.1 - /opt/homebrew/bin/pod
  SDKs:
    iOS SDK:
      Platforms: DriverKit 22.2, iOS 16.2, macOS 13.1, tvOS 16.1, watchOS 9.1
    Android SDK:
      API Levels: 26, 28, 29, 30, 31, 32, 33
      Build Tools: 28.0.3, 29.0.2, 29.0.3, 30.0.2, 30.0.3, 31.0.0, 33.0.0, 33.0.2
      System Images: android-33 | Google APIs ARM 64 v8a
      Android NDK: Not Found
  IDEs:
    Android Studio: Flamingo 2022.2.1 Patch 2 Flamingo 2022.2.1 Patch 2
    Xcode: 14.2/14C18 - /usr/bin/xcodebuild
  Languages:
    Java: 11.0.16.1 - /usr/bin/javac
  npmPackages:
    @react-native-community/cli: Not Found
    react: 18.0.0 => 18.0.0
    react-native: 0.69.12 => 0.69.12
    react-native-macos: Not Found
  npmGlobalPackages:
    *react-native*: Not Found
```

--- a/packages/core/android/build.gradle
+++ b/packages/core/android/build.gradle
@@ -119,3 +119,13 @@ dependencies {
 
     implementation "com.airbnb.android:lottie:6.0.0"
 }
+
+if (isNewArchitectureEnabled()) {
+  // rncli & react-native-gradle-plugin use codegenConfig in package.json since react-native@0.70
+  // Use react{} for compatibility with react-native@0.68/0.69
+  react {
+    jsRootDir = file("../src/specs")
+    libraryName = 'lottiereactnative'
+    codegenJavaPackageName = "com.airbnb.android.react.lottie"
+  }
+}

